pkg/testmachinery/controller/admission/webhooks/validator.go

```diff
@@ -48,6 +94,13 @@ func (v *testrunValidator) InjectDecoder(d *admission.Decoder) error {
 
 // Handle validates a testrun
 func (v *testrunValidator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	healthMutex.Lock()
+	if healthErr != nil {
+		defer healthMutex.Unlock()
+		return admission.Denied(healthErr.Error())
```

Comment:
I think we should rather error with `admission.Errored(http.InternalError, err)`.
Maybe something more suitable than `internalError`.
Because the testrun itself is not necessary wrong. but we some internal dependency errors

Reply:
HTTP 424 seems to be suitable.